src/ldp.rs

```diff
@@ -53,6 +53,120 @@ lazy_static! {
     };
 }
 
+pub fn get_proof_suite(proof_type: &str) -> Result<&(dyn ProofSuite + Sync), Error> {
+    Ok(match proof_type {
+        "RsaSignature2018" => &RsaSignature2018,
+        "Ed25519Signature2018" => &Ed25519Signature2018,
+        "Ed25519BLAKE2BDigestSize20Base58CheckEncodedSignature2021" => {
+            &Ed25519BLAKE2BDigestSize20Base58CheckEncodedSignature2021
+        }
+        "P256BLAKE2BDigestSize20Base58CheckEncodedSignature2021" => {
+            &P256BLAKE2BDigestSize20Base58CheckEncodedSignature2021
+        }
+        "EcdsaSecp256k1Signature2019" => &EcdsaSecp256k1Signature2019,
+        "EcdsaSecp256k1RecoverySignature2020" => &EcdsaSecp256k1RecoverySignature2020,
+        "Eip712Signature2021" => {
+            #[cfg(not(feature = "keccak-hash"))]
+            return Err(Error::MissingFeatures("keccak-hash"));
+            #[cfg(feature = "keccak-hash")]
+            &Eip712Signature2021
+        }
+        "EthereumPersonalSignature2021" => {
+            #[cfg(not(feature = "keccak-hash"))]
+            return Err(Error::MissingFeatures("keccak-hash"));
+            #[cfg(feature = "keccak-hash")]
+            &EthereumPersonalSignature2021
+        }
+        "EthereumEip712Signature2021" => {
+            #[cfg(not(feature = "keccak-hash"))]
+            return Err(Error::MissingFeatures("keccak-hash"));
+            #[cfg(feature = "keccak-hash")]
+            &EthereumEip712Signature2021
+        }
+        "TezosSignature2021" => &TezosSignature2021,
+        "SolanaSignature2021" => &SolanaSignature2021,
+        "JsonWebSignature2020" => &JsonWebSignature2020,
+        "EcdsaSecp256r1Signature2019" => &EcdsaSecp256r1Signature2019,
+        _ => return Err(Error::ProofTypeNotImplemented),
+    })
+}
+
+fn pick_proof_suite<'a, 'b>(
```

Comment:
Should the two functions be merged in a single method/trait. With the `pick` version being just the default in case there isn't one specified?